### PR TITLE
Editing/removing of attached pictures is now possible via web

### LINKED ISSUE #12827: Can't delete images from posts
### Bug Description

When doing post editing through the UI it is not possible to actually delete images from the post. If one removes the [url] itself the text is no longer in the post text but the images are still associated with it and they continue to be rendered. At that point there is no way to remove it at all either.

### Steps to Reproduce
1. Create a post with two images on it
2. Remove one of the two images in post editing by removing the attachment data
3. Add a new "second" image

Actual Result:

Post text reads:
```
post with two images eueeue
[url=https://friendica.myportal.social/photos/hankg/image/163237620862a1652e06bd0520719010][img=https://friendica.myportal.social/photo/163237620862a1652e06bd0520719010-1.png][/img][/url]

[url=https://friendica.myportal.social/photos/hankg/image/939459848163d5b8578de1d347371344][img=https://friendica.myportal.social/photo/939459848163d5b8578de1d347371344-1.jpg]Screenshot of “Nothing Breaks Like a Heart” single cover by Mark Ronson and Miley Cyrus.[/img][/url]
```

But generates the post with all three images. Before adding the third it rendered with the original two despite the text of the post no longer containing it

image

Expected Result:

Post text seems consistent with the edits but the attachments need to reflect that one was deleted. When rendering the edited post we need to difference the new attachment list with the old and delete the old attachment references in post-media table.

### Platform Info
Tested with the latest head of the develop branch as well as on my production server:

This is Friendica, version 2023.01 that is running at the web location https://friendica.myportal.social. The database version is 1502/1502, the post update version is 1484/1484.

### PR DESCRIPTION
**Pull request by annando**
This fixes some parts of #12415 and fixes #12827, since it is now possible to edit attached pictures via web.

### CODE REVIEW

File: src/Model/Post/Media.php [hunk 1]
```diff
@@ -584,9 +590,14 @@ public static function insertFromRelevantUrl(int $uriid, string $body, string $f
...
-				self::insert(['uri-id' => $uriid, 'type' => self::UNKNOWN, 'url' => $url], false, $network);
-				if ($network == Protocol::DFRN) {
+				$result = self::insert(['uri-id' => $uriid, 'type' => self::UNKNOWN, 'url' => $url], false, $network);
+				if ($result && ($network == Protocol::DFRN)) {
 					self::revertHTMLType($uriid, $url, $fullbody);
+					Logger::debug('Revert HTML type', ['uri-id' => $uriid, 'url' => $url]);
+				} elseif ($result) {
+					Logger::debug('Media had been added', ['uri-id' => $uriid, 'url' => $url]);
+				} else {
+					Logger::debug('Media had not been added', ['uri-id' => $uriid, 'url' => $url]);
```
Comment (MrPetovan) [inline src/Model/Post/Media.php]:
```suggestion
					Logger::debug('Media has been added', ['uri-id' => $uriid, 'url' => $url]);
				} else {
					Logger::debug('Media has not been added', ['uri-id' => $uriid, 'url' => $url]);
```

**Review by MrPetovan — CHANGES_REQUESTED** [review]
Small language fix.

File: src/Model/Post/Media.php (hunk 1 above)
Comment (annando) [reply src/Model/Post/Media.php]: Nitpicker 😁

**Review by MrPetovan — APPROVED** [review]

File: src/Model/Post/Media.php (hunk 1 above)
Comment (MrPetovan) [reply src/Model/Post/Media.php]: Not everyone can understand Deutschglish as well as myself!

**Pull request merged**